oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexEditorProvider.java

```diff
@@ -257,6 +261,22 @@ public Editor getIndexEditor(
         return null;
     }
 
+    private void initDefaultWriterFactory() {
+        defaultWriterFactoryInitLock.lock();
+        try {
+            if (defaultIndexWriterFactory == null) {
+                LOG.info("Initializing DefaultIndexWriterFactory");
+                cowDirectoryCleanupCallback = new COWDirectoryCleanupCallback();
+                BlobDeletionCallback blobDeletionCallback = activeDeletedBlobCollector.getBlobDeletionCallback();
```

Comment:
I wonder if it is possible to add a test case to show the problem of the current code